File: GitToolBox/src/zielu/gittoolbox/cache/PerRepoStatusCacheListener.java
package zielu.gittoolbox.cache;

import com.google.common.collect.ImmutableMap;
import git4idea.repo.GitRepository;
import org.jetbrains.annotations.NotNull;

public interface PerRepoStatusCacheListener {
    void initialized(ImmutableMap<GitRepository, RepoInfo> info);
    void stateChanged(@NotNull RepoInfo info,  @NotNull GitRepository repository);
}

File: src/main/kotlin/zielu/gittoolbox/cache/PerRepoStatusCacheListener.kt
package zielu.gittoolbox.cache

import git4idea.repo.GitRepository

internal interface PerRepoStatusCacheListener {
  @JvmDefault
  fun stateChanged(previous: RepoInfo, current: RepoInfo, repository: GitRepository) {
    stateChanged(current, repository)
  }

  @JvmDefault
  fun stateChanged(info: RepoInfo, repository: GitRepository) {}

  @JvmDefault
  fun evicted(repositories: Collection<GitRepository>) {}

  @JvmDefault
  fun allRepositoriesInitialized(repositories: Collection<GitRepository>) {}
}
